sale_product_variant_attribute_tax/tests/test_sale_product_variant_attribute_tax.py

```diff
@@ -57,10 +58,29 @@ def setUpClass(cls):
                 ],
             }
         )
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": cls.product_template.name,
+                "list_price": 100,
+                "product_tmpl_id": cls.product_template.id,
+                "product_attribute_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_tmpl_id": cls.product_template.id,
+                            "attribute_id": cls.attribute.id,
+                            "value_id": cls.attribute_value.id,
+                            "owner_model": "sale.order.line",
+                        },
+                    )
+                ],
+            }
+        )
```

Comment:
This is not necessary, you can remove `"no_create_variants": "yes"` first.

It is important to remember the behavior of `sale_variant_configurator`, do NOT manually define `product_id`, we define a template (`product_tmpl_id`) and attribute values and the `product_id` field will define itself.

Reply:
Exactly.